meta-oe/recipes-core/sdbus-c++/sdbus-c++_0.7.2.bb

```diff
@@ -0,0 +1,46 @@
+SUMMARY = "sdbus-c++"
+DESCRIPTION = "High-level C++ D-Bus library designed to provide easy-to-use yet powerful API in modern C++"
+
+SECTION = "libs"
+
+LICENSE = "LGPLv2.1"
+LIC_FILES_CHKSUM = "file://${WORKDIR}/sdbus-cpp-${PV}/COPYING;md5=1803fa9c2c3ce8cb06b4861d75310742"
+
+inherit cmake pkgconfig systemd ptest python3native
+
+DEPENDS += "expat ${@bb.utils.contains('DISTRO_FEATURES', 'systemd', 'systemd', 'meson-native ninja-native git-native gperf-native gettext-native util-linux glibc libcap', d)}"
+RDEPENDS_${PN} = "${@bb.utils.contains('DISTRO_FEATURES', 'systemd', 'libsystemd', 'glibc libcap', d)}"
+
+SRCREV = "750dab39270f338181db8d841cd4b1a9a50ae4d7"
+SRC_URI = "git://github.com/Kistler-Group/sdbus-cpp.git;protocol=https;destsuffix=sdbus-cpp-${PV};branch=master"
+SRC_URI += "file://run-ptest"
+
+TEST_BIN_DIR = "/opt/test/bin"
+
+# Allow CMake to use git on the host machine
+OECMAKE_FIND_ROOT_PATH_MODE_PROGRAM = "BOTH"
+# Ninja is the default CMake generator in Yocto now, so let's set Unix Makefiles explicitly
```

Comment:
Please say why, are the generated Makefiles incompatible with Ninja?

Reply:
When building the recipe with Ninja, I get: `ninja: error: 'libsystemd-v242/src/LibsystemdBuildProject-build/libsystemd.a', needed by 'libsdbus-c++.so.0.7.2', missing and no known rule to make it`.

sdbus-c++ uses ExternalProject feature of CMake and builds libsystemd using that. This is where it doesn't work with Ninja out of the box.